org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/GradleLaunchConfigurationManager.java

```diff
@@ -33,14 +40,25 @@
     /**
      * Returns either a new Gradle {@link ILaunchConfiguration} instance or an existing one,
      * depending on whether there is already a Gradle run configuration for the given set of
-     * attributes or not.
+     * attributes or not. The result is saved to the disk.
      *
      * @param configurationAttributes the run configuration attributes, must not be not null
      * @return the new or existing Gradle run configuration
      */
     ILaunchConfiguration getOrCreateRunConfiguration(GradleRunConfigurationAttributes configurationAttributes);
 
     /**
+     * Returns either a new Gradle {@link ILaunchConfiguration} instance or an existing one,
+     * depending on whether there is already a Gradle run configuration for the given set of
+     * attributes or not.
+     *
+     * @param configurationAttributes the run configuration attributes, must not be not null
+     * @param saveStrategy determines whether the new run configuration should be saved to the disk.
+     * @return the new or existing Gradle run configuration
+     */
+    ILaunchConfiguration getOrCreateRunConfiguration(GradleRunConfigurationAttributes configurationAttributes, SaveStrategy saveStrategy);
```

Comment:
I had the idea that we should use `ILaunchConfiguration` for all task executions. This was not the case previously when a test was executed via a build operation. But in that case, we shouldn't save the run configuration as the build operation is not persistable. So I added the method above to create a run config without actually saving it.